test/e2e/route_advertisements.go

```diff
@@ -463,3 +472,397 @@ spec:
+					framework.ExpectNoError(err)
+					return podsNetA[2].Name, podsNetA[2].Namespace, net.JoinHostPort(statusPodB.IPs[ipIndex].IP.String(), "8080") + "/clientip", curlConnectionTimeoutCode, true
+				}),
+
+			ginkgo.Entry("pod to pod on different networks and different nodes not should work",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					// podsNetA[0] and podNetB are on different nodes
+					statusPodB, err := userDefinedNetworkStatus(podNetB, namespacedName(podNetB.Namespace, cudnBTemplate.Name))
+					framework.ExpectNoError(err)
+					// FIXME: L3 - pod to pod on different networks and different nodes should NOT work
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(statusPodB.IPs[ipIndex].IP.String(), "8080") + "/clientip", "", false
+				}),
+			ginkgo.Entry("pod in the default network should not be able to access a UDN service",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					return podNetDefault.Name, podNetDefault.Namespace, net.JoinHostPort(svcNetA.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", curlConnectionTimeoutCode, true
+				}),
+			ginkgo.Entry("pod in the UDN should be able to access a service in the same network",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(svcNetA.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", "", false
+				}),
+			ginkgo.Entry("pod in the UDN should not be able to access a default network service",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					err := true
+					out := curlConnectionTimeoutCode
+					if isLocalGWModeEnabled() {
+						// FIXME: L3 - pod in the UDN should NOT be able to access a default network service
+						err = false
+						out = ""
+					}
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(svcNetDefault.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", out, err
+				}),
+			ginkgo.Entry("pod in the UDN should not be able to access a service in a different UDN",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(svcNetB.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", curlConnectionTimeoutCode, true
+				}),
+			ginkgo.Entry("host to a local UDN pod should not work",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					statusPodA, err := userDefinedNetworkStatus(podsNetA[0], namespacedName(podsNetA[0].Namespace, cudnATemplate.Name))
+					framework.ExpectNoError(err)
+					// FIXME: L3 - curl returns code 7, the request works partially
```

Comment:
HEH
this is the piece we spoke about with how vrf's work?
I don't think we will fix this now but I think for history purposes and because I am selfish I want to remember it do you want to replace it with why those partial conn works for the future-pat and future-surya 4 months from now? :D

and then FIXME: will be that your fix PR will fix this